### PR TITLE
Add a warning in case of incorrect syntax of highlighting style.

### PR DESCRIPTION
**Pull request by aawad6**
### Add a warning in case of incorrect syntax of highlighting style (without `=`).
This PR is related to [#issue1545](https://github.com/apache/logging-log4j2/issues/1545),  it meant to ensure that the user is providing a correct syntax for highlighting style.

e.g. `%highlight{%5level}{LOGBACK}` should be `%highlight{%5level}{STYLE=LOGBACK}`. For more information see the documentation [here](https://logging.apache.org/log4j/2.x/manual/layouts.html#PatternLayout).

### CODE REVIEW

**Comment by aawad6** [comment]
@ppkarwasz I wonder If there's a way to check on a specific logged message in one of the unit tests?

**Review by garydgregory — CHANGES_REQUESTED** [review]
Better message IMO: 
`Syntax error, missing '=': Expected "{KEY1=VALUE, KEY2=VALUE, ...}"`

**Comment by aawad6** [comment]
Noted, but I have a question regarding the following code block:
```
final String string = options[1]
              .replaceAll(PatternParser.DISABLE_ANSI + "=(true|false)", Strings.EMPTY)
              .replaceAll(PatternParser.NO_CONSOLE_NO_ANSI + "=(true|false)", Strings.EMPTY);
```
To avoid false positive warning messages, aren't we need to replace also the commas with Strings.empty()?
for example: if  `Options[1] = "noConsoleNoAnsi=false, disableAnsi=false, LOGBACK"`, `string` will be `", , LOGBACK"`.

So I'd suggest to tweak the match pattern to looks like the following:
```
final String string = options[1]
        .replaceAll(",?" + PatternParser.DISABLE_ANSI + "=(true|false),?", Strings.EMPTY)
        .replaceAll(",?" + PatternParser.NO_CONSOLE_NO_ANSI + "=(true|false),?", Strings.EMPTY)
        .trim();
```

**Comment by ppkarwasz** [comment]
@aawad6,

A while back I added a `@UsingStatusListener` JUnit 5 extensions, which allows you to capture status logger messages into a `ListStatusListener`:
```java
    @Test
    @UsingStatusListener
    public void testSomething(final ListStatusListener listener) {
        ...
        assertThat(listener.findStatusData(Level.WARN)).hasSize(1);
        ...
    }
```

Regarding the "disableAnsi" and "noConsoleNoAnsi" keys, we could probably deal with them better. E.g. instead of removing them through a regex, we can add them to the `dontEscapeKeys` and deal with them in the `HighlightConverter` constructor.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/HighlightConverter.java [hunk 1]
```diff
@@ -87,6 +87,10 @@ public final class HighlightConverter extends LogEventPatternConverter implement
 
     private static final String STYLE_KEY = "STYLE";
 
+    private static final String DISABLE_ANSI_KEY = "DISABLEANSI";
+
+    private static final String NO_CONSOLE_NO_ANSI_KEY = "NOCONSOLENOANSI";
+
```
Comment (ppkarwasz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/HighlightConverter.java:93]: These are already defined as `PatternParser.DISABLE_ANSI` and `PatternParser.NO_CONSOLE_NO_ANSI`.
They differ in casing, but keys are matched case-insensitive anyway.

**Review by ppkarwasz — APPROVED** [review]
Apart from the comment below, it looks good to me.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/HighlightConverter.java (hunk 1 above)
Comment (aawad6) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/HighlightConverter.java:92]: I think we still need the UPPERCASE keys for both `disableAnsi `and `noConsoleNoAnsi`. 

The issue arises when the `key` is obtained and transformed to UPPERCASE characters. Next, search for that key in the `dontEscapeKeys` array. If the `key` was previously supplied as lowercase, the `escape` variable will be <0, which means the `value` was overridden.
```
final String key = toRootUpperCase(keyValue[0]);
final String value = keyValue[1];
final boolean escape = Arrays.binarySearch(sortedIgnoreKeys, key) < 0;
map.put(key, escape ? createSequence(value.split("\\s")) : value);
```

Comment (ppkarwasz) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/HighlightConverter.java:92]: I missed that, sorry.

**Comment by vy** [comment]
Thanks for the improvement @aawad6, much appreciated! :bow: Merging it...

**Pull request merged**